src/java.base/share/classes/java/time/ZoneOffset.java

```diff
@@ -424,11 +424,15 @@ public static ZoneOffset ofTotalSeconds(int totalSeconds) {
             throw new DateTimeException("Zone offset not in valid range: -18:00 to +18:00");
         }
         if (totalSeconds % (15 * SECONDS_PER_MINUTE) == 0) {
-            return SECONDS_CACHE.computeIfAbsent(totalSeconds, totalSecs -> {
-                ZoneOffset result = new ZoneOffset(totalSecs);
+            Integer totalSecs = totalSeconds;
+            ZoneOffset result = SECONDS_CACHE.get(totalSecs);
+            if (result == null) {
+                result = new ZoneOffset(totalSeconds);
+                SECONDS_CACHE.putIfAbsent(totalSecs, result);
+                result = SECONDS_CACHE.get(totalSecs);
```

Comment:
putIfAbsent() returns the existing value (if any), the 2nd get can be avoided by using the non-null value if returned or returning the result just created.
```suggestion
                var existing = SECONDS_CACHE.putIfAbsent(totalSecs, result);
                return (existing != null) ? existing : result;
```
It should perform better than doing the `get` again.

Reply:
That's right. Committed as suggested